### PR TITLE
MF-998 - Add Twins service to Makefile and docker-compose.yml

### PR DESCRIPTION
**Pull request by darkodraskovic**
Signed-off-by: Darko Draskovic <darko.draskovic@gmail.com>

### CODE REVIEW

**Review by manuio — DISMISSED** [review]
LGTM

File: docker/docker-compose.yml [hunk 1]
```diff
@@ -263,3 +265,47 @@ services:
       - ${MF_COAP_ADAPTER_PORT}:${MF_COAP_ADAPTER_PORT}/tcp
     networks:
       - mainflux-base-net
+
+  twins-db:
+    image: mongo:3.6-jessie
```
Comment (drasko) [inline docker/docker-compose.yml]: Although I have nothing against, I would advise using latest stable - 4.2: https://docs.mongodb.com/manual/release-notes/

Also, it looks like for some reason they prefer Ubuntu (Bionic) then Debian Jessie by default: https://hub.docker.com/_/mongo

File: .env [hunk 2]
```diff
@@ -196,3 +196,20 @@ MF_POSTGRES_READER_DB_SSL_MODE=disable
...
+MF_TWINS_SERVER_KEY=""
+MF_TWINS_DB_NAME=mainflux
+MF_TWINS_DB_HOST=twins-db
+MF_TWINS_DB_PORT=27017
+MF_TWINS_SINGLE_USER_EMAIL=""
+MF_TWINS_SINGLE_USER_TOKEN=""
+MF_TWINS_CLIENT_TLS=""
+MF_TWINS_CA_CERTS=""
+MF_TWINS_MQTT_URL=tcp://mqtt-adapter:1883
+MF_TWINS_THING_ID=cfc93379-d09a-438b-806e-e3cda2180124
```
Comment (drasko) [inline .env]: Do we need these default values, or is it better to leave this empty?

**Review by drasko — CHANGES_REQUESTED** [review]

File: docker/docker-compose.yml (hunk 1 above)
Comment (darkodraskovic) [reply docker/docker-compose.yml]: Will be fixed.

File: .env (hunk 2 above)
Comment (darkodraskovic) [reply .env]: Unfotunatelly, these values are necessary. Therefore, we should leave them empty. What is more, I will move twins service to addons.

Comment (darkodraskovic) [reply .env]: I'm working towards eliminating these env vars alltogether.

File: docker/nginx/entrypoint.sh [hunk 3]
```diff
@@ -14,6 +14,7 @@ envsubst '
     ${MF_THINGS_HTTP_PORT}
     ${MF_THINGS_HTTP_PORT}
     ${MF_HTTP_ADAPTER_PORT}
+    ${MF_TWINS_HTTP_PORT}
```
Comment (manuio) [inline docker/nginx/entrypoint.sh]: Remove this

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by manuio — DISMISSED** [review]
LGTM

File: .env (hunk 2 above)
Comment (mteodor) [reply .env]: some vars are not used in docker-compose

**Review by mteodor — DISMISSED** [review]

File: docker/addons/twins/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  docker_mainflux-base-net:
+    external: true
+
+volumes:
+  mainflux-twins-db-volume:
+  mainflux-twins-db-configdb-volume:
+
+services:
+  twins-db:
+    image: mongo:3.6-jessie
```
Comment (drasko) [inline docker/addons/twins/docker-compose.yml]: Use new Mongo release, test with it. Use Ubuntu, as they switched to Ubuntu (Bionic I think).

File: .env (hunk 2 above)
Comment (drasko) [reply .env]: Is this still ongoing?

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

File: docker/addons/twins/docker-compose.yml (hunk 4 above)
Comment (drasko) [reply docker/addons/twins/docker-compose.yml]: Is there a particular reason for sticking with old MondoDB release?

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**